src/app/scripts/cac/pages/cac-pages-map.js

```diff
@@ -0,0 +1,69 @@
+CAC.Pages.Map = (function ($, MapControl, MapRouting, MockDestinations) {
+    'use strict';
+
+    var defaults = {
+        map: {}
+    };
+    var mapControl = null;
+
+    function Map(options) {
+        this.options = $.extend({}, defaults, options);
+    }
+
+    Map.prototype.initialize = function () {
+
+        // Map initialization logic and event binding
+        mapControl = new MapControl();
+        mapControl.plotLocations(MockDestinations);
+        mapControl.locateUser();
+
+        // Register events to catch location clicks and autocomplete direction dialog
+        $(document).bind('MOS.Map.Control.CurrentLocationClicked', function(e, lat, lng) {
+            var coords = lat + ',' + lng;
+            $('section.directions input.origin').val(coords);
+        });
+        $(document).bind('MOS.Map.Control.DestinationClicked', function(e, feature) {
+            var coords = feature.geometry.coordinates[1] + ',' + feature.geometry.coordinates[0];
+            $('section.directions input.destination').val(coords);
+        });
+
+        // Plan a trip using information provided
+        $('section.directions button[type=submit]').click(function() {
```

Comment:
Consider breaking some of the anonymous functions here into named private functions. To keep `this` bound to the instance of Pages.Map, you can do:
```
$('selector').click($.proxy(foo, this));

function foo() { ... };
```
See http://api.jquery.com/jquery.proxy/